proposals/lip-0043.md

```diff
@@ -280,75 +275,65 @@ This command does not have an extra fee, i.e., `extra fee = 0`.
+                        "fieldNumber": 2
+                    }
+                }
+            }
         },
-      }
-    },
-    "signature": {
-      "dataType": "bytes",
-      "fieldNumber": 4
+        "signature": {
+            "dataType": "bytes",
+            "fieldNumber": 4
+        },
+        "aggregationBits": {
+            "dataType": "bytes",
+            "fieldNumber": 5
+        }
     },
-    "aggregationBits": {
-      "dataType": "bytes",
-      "fieldNumber": 5
-    }
-  },
-  "required": [
-    "ownChainID",
-    "ownName",
-    "mainchainValidators",
-    "signature",
-    "aggregationBits"
-  ]
+    "required": [
+        "ownChainID",
+        "ownName",
+        "mainchainValidators",
+        "signature",
+        "aggregationBits"
+    ]
 }
 ```
 
 #### Verification
 
 Let `trs` be a transaction with module ID `MODULE_ID_INTEROPERABILITY` and command ID `COMMAND_ID_MAINCHAIN_REG` to be verified. The set of validity rules to validate `trs.params` are:
 
-* The `trs.params.mainchainValidators.keys` array must contain 101 elements (exact number of active delegates on Lisk mainchain.
-* Every element in the `trs.params.mainchainValidators.keys` array has to be unique.
-* Elements in the `trs.params.mainchainValidators.keys` array have a length of 48 bytes.
-* The `trs.params.mainchainValidators.keys` array must be ordered lexicographically.
-* The `trs.params.mainchainValidators.weights` array must contain the same number of elements as `trs.params.mainchainValidators.keys`.
-* The elements in the `trs.params.mainchainValidators.weights` array are integers with value equal to 1.
-* The value of the `trs.params.mainchainValidators.certificateThreshold` property has to be 68.
+* The `trs.params.ownChainID` property has to be an integer less than or equal to `MAX_UINT32`. 
+* The `trs.params.ownName` property has to contain only characters from the set `[a-z0-9!@$&_.]` and has to be at most `MAX_LENGTH_NAME` characters long.
+* Rules for `trs.params.mainchainValidators` array:
+    *   The array must contain 101 elements (exact number of active delegates on Lisk mainchain).
+    *   The array must be ordered lexicographically by `blsKey` property.
+    *   The `blsKey` property of each element has a length of 48 bytes.
+    *   The `blsKey` property of each element is unique within the array.
+    *   The `bftWeight` property of each element is equal to 1.
 * The properties `trs.params.aggregationBits` and `trs.params.signature` are validated as follows:
   * The function `verifyWeightedAggSig(keyList, tag, netID, aggregationBits, signature, weights, certificateThreshold, message)` specified in the [LIP 0038][BLSkeys], must return `VALID` where:
     * The `keyList` property is an array containing `getValidatorAccount(address).blsKey` for every address in `currentValidators = getGeneratorList()`, where `getValidatorAccount` and `getGeneratorList` are functions exposed by the [validators module][validatorsModule].
```

Comment:
I think it would be better to use the active validators defined in the BFT module here as there is not necessarily a 1-1 correspondence between the current generator list and the active validators set in the BFT module (e.g., due to standby delegates). Further, the API of the BFT module needs to be updated to use `getBFTParameters` as `getValidatorBFTWeights` and `getThresholds` do not exist any more.

Reply:
good points, I updated this to make them consisten with the new specs in the BFT module. I am not very sure about the input `currentHeight` for `getBFTParameters` in this context though, can you confirm it is right?

Reply:
I would use the height of the current block, see https://github.com/LiskHQ/lips/pull/109#discussion_r709113164